library/signed/src/Utils.qs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// Util functions for internal use by the signed integer library.
+
+import Std.Math.Min;
+import Std.Diagnostics.Fact;
+import Std.Arrays.Head, Std.Arrays.Tail, Std.Arrays.Most, Std.Arrays.Rest;
+
+operation AndLadder(controls : Qubit[], targets : Qubit[]) : Unit is Adj {
+    let controls1 = [Head(controls)] + Most(targets);
+    let controls2 = Rest(controls);
+    for (a, b, c) in Zipped3(controls1, controls2, targets) {
+        AND(a, b, c);
+    }
+}
+
+function Zipped3<'T1, 'T2, 'T3>(first : 'T1[], second : 'T2[], third : 'T3[]) : ('T1, 'T2, 'T3)[] {
```

Comment:
We try to avoid using Zipped function, when possible, for faster simulation speed. Both AndLadder and ApplyCCNOTChain can be re-written using a simple loop.